packages/utils/pack-up/src/node/tasks/vite/config.ts

```diff
@@ -120,14 +122,7 @@ const resolveViteConfig = (ctx: BuildContext, task: ViteBaseTask) => {
         },
       },
     },
-    /**
-     * We _could_ omit this, but we'd need to introduce the
-     * concept of a custom config for the scripts straight away
-     *
-     * and since this is isolated to the Strapi CLI, we can make
-     * some assumptions and add some weight until we move it outside.
-     */
-    plugins: runtime === 'node' ? [] : [react()],
+    plugins: [...basePlugins, ...ctx.plugins],
```

Comment:
I'm wondering if it would be better to make this a function that receives e.g. `runtime` as argument, because I'd imagine every time you add a plugin you need to define for which runtime it applies?

Reply:
That's a nice idea. You mean make it a function in the config? it could be a func or array:

```
plugins: PluginOption[] | (runtime: Runtime) => PluginOption[]
```

Reply:
yep, that sounds nice. Maybe `{ runtime }` instead as argument to make it easier to extend.

Reply:
![Scrambled Eggs Reaction GIF](https://media1.giphy.com/media/S9FtQO5fxOMsT5l1wa/200.gif)